src/function/scalar/date/date_trunc.cpp

```diff
@@ -0,0 +1,226 @@
+#include "duckdb/function/scalar/date_functions.hpp"
+#include "duckdb/common/enums/date_trunc_specifier.hpp"
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/types/date.hpp"
+#include "duckdb/common/types/timestamp.hpp"
+#include "duckdb/common/string_util.hpp"
+
+// TODO date_trunc function should also handle interval data type when it is implemented. See
+// https://www.postgresql.org/docs/9.1/functions-datetime.html
+
+using namespace std;
+
+namespace duckdb {
+
+DateTruncSpecifier GetDateTruncSpecifier(string specifier) {
```

Comment:
It seems like this is mostly shared with date_part and the DatePartSpecifier, perhaps we could merge them together?

Reply:
Sure!